--- ./patchsource/www/rega/esp/functions.fn.orig	2019-02-23 12:13:20.000000000 +0100
+++ ./patchsource/www/rega/esp/functions.fn	2019-04-23 21:11:37.000000000 +0200
@@ -331,6 +331,13 @@
     ! Hide channel 3 of this device
     bShowChannel = false;
   }
+  
+  ! HB Status Display EPaper 42BW
+  if(( (oDevHssType == "HB-DIS-EP-42BW") || (oDevHssType == "HB-DIS-EP-42BW-MAINS") ) && (chNumber == 11)) {
+    ! Hide channel 11 of this device
+    bShowChannel = false;
+  }
+  
 }
 
 function ::showHideHmIPChannel() {
