--- ./patchsource/www/webui/webui.js.orig	2019-02-23 12:14:40.000000000 +0100
+++ ./patchsource/www/webui/webui.js	2019-02-23 12:37:16.000000000 +0100
@@ -13664,7 +13664,7 @@
   // Check each line if the free text mode is active.
   // If yes switch on the appropriate text input field
   // and set the variable freeTextActive to true
-  for (var line = 0; line < 6; line++) {
+  for (var line = 0; line < 10; line++) {
     var jLine = jQuery("#textSelect_" + line),
     jCellFreeText = jQuery("#cellFreeText_"+line);
 
@@ -13680,7 +13680,7 @@
   // If at least one line has an active free text mode
   // we have to activate the placeholder for the other lines and the header (color and icon are moving up)
   if (freeTextActive) {
-    for (var line = 0; line < 6; line++) {
+    for (var line = 0; line < 10; line++) {
       var jLine = jQuery("#textSelect_" + line),
       jPlaceHolder = jQuery("#placeHolder_"+line);
 
@@ -13692,7 +13692,7 @@
     }
   } else {
     jHeadFreeText.hide().css("display", "none");
-    for (var line = 0; line < 6; line++) {
+    for (var line = 0; line < 10; line++) {
       jQuery("#placeHolder_"+line).hide().css("display", "none");
     }
   }
