--- ./patchsource/www/rega/esp/functions.fn.orig
+++ ./patchsource/www/rega/esp/functions.fn
@@ -326,6 +326,18 @@
     bShowChannel = false;
   }
 
+  ! HB Status Display EPaper 42BW
+  if(( (oDevHssType == "HB-DIS-EP-42BW") || (oDevHssType == "HB-DIS-EP-42BW-MAINS") ) && (chNumber == 11)) {
+    ! Hide channel 11 of this device
+    bShowChannel = false;
+  }
+
+  ! HB Status Display EPaper 75BW
+  if(( (oDevHssType == "HB-DIS-EP-75BW") || (oDevHssType == "HB-DIS-EP-75BW-MAINS") ) && (chNumber == 9)) {
+    ! Hide channel 9 of this device
+    bShowChannel = false;
+  }
+  
   ! Status Display EPaper
   if(( (oDevHssType == "HM-Dis-EP-WM55") ) && (chNumber == 3)) {
     ! Hide channel 3 of this device
