--- ./patchsource/www/rega/esp/side.inc.orig	2019-02-23 12:13:34.000000000 +0100
+++ ./patchsource/www/rega/esp/side.inc	2019-02-23 12:25:44.000000000 +0100
@@ -12,9 +12,11 @@
   string hm_oucfm = "HM-OU-CFM-Pl";
   string hm_oucfmTW = "HM-OU-CFM-TW";
   string hm_oucmpcb = "HM-OU-CM-PCB";
+  string hb_oump3led = "HB-OU-MP3-LED";
   string hm_partyDialog = "HM-CC-RT-DN HM-TC-IT-WM-W-EU HM-CC-VG-1";
   string hm_statusDisplay = "HM-Dis-WM55";
   string hm_statusDisplayEPaper = "HM-Dis-EP-WM55";
+  string hb_statusDisplayEPaper = "HB-DIS-EP-42BW";
   string hm_rgbw_controller = "HM-LC-RGBW-WM";
   string hm_jalousieActor = "HM-LC-Ja1PBU";
   string heatingClimateControlTransceiver =  "HEATING_CLIMATECONTROL_TRANSCEIVER";
@@ -451,9 +453,11 @@
             integer oucfm = chLabel.Find(hm_oucfm);
             integer oucfmTW = chLabel.Find(hm_oucfmTW);
             integer oucmpcb = chLabel.Find(hm_oucmpcb);
+            integer oump3led = chLabel.Find(hb_oump3led);
             integer partyDialogDevice = hm_partyDialog.Find(chLabel);
             integer statusDisplayDevice = hm_statusDisplay.Find(chLabel);
             integer statusDisplayEPaperDevice = hm_statusDisplayEPaper.Find(chLabel);
+            integer statusHbDisplayEPaperDevice = hb_statusDisplayEPaper.Find(chLabel);
             integer rgbw_controller = chLabel.Find(hm_rgbw_controller);
             integer i_vir_lg_rgb_dim = chLabel.Find(s_vir_lg_rgb_dim);
             integer i_vir_lg_rgbw_dim = chLabel.Find(s_vir_lg_rgbw_dim);
@@ -462,6 +466,7 @@
 
             if ((statusDisplayDevice != -1)
               || (statusDisplayEPaperDevice != -1)
+              || (statusHbDisplayEPaperDevice != -1)
               || (rgbw_controller != -1)
               || ((jalousieActor != - 1) && (oDP.HSSID().Find("LEVEL_COMBINED") != -1))
               || (((i_vir_lg_rgb_dim != - 1) && (oDP.HSSID().Find("RGB") != -1)) || ((i_vir_lg_rgbw_dim != - 1) && (oDP.HSSID().Find("RGBW") != -1)) || (( i_vir_lg_group != - 1) && (oDP.HSSID().Find("RGBW") != -1)))
@@ -482,6 +487,13 @@
                 Write( '<input id="prgStringOptionValue'#oSD.ID()#'" onfocus="SetStatusDisplay('#oSD.ID()#',\'DIS-EP\');" type="text" class="SelectBox" size="10" value="'#iV#'" onchange="iseSingleDestination.SetValue('#oSD.ID()#',this.value,\'STRING\');" /> ' );
               }
 
+              if (statusHbDisplayEPaperDevice != -1) {
+                if(iV == "1") {
+                  ! This is the default string for 'all values not used'
+                  iV = "0x02,0x0A,0x0A,0x0A,0x0A,0x0A,0x0A,0x0A,0x0A,0x0A,0x0A,0x03";
+                }
+                Write( '<input id="prgStringOptionValue'#oSD.ID()#'" onfocus="SetStatusDisplay('#oSD.ID()#',\'DIS-EP42BW\');" type="text" class="SelectBox" size="10" value="'#iV#'" onchange="iseSingleDestination.SetValue('#oSD.ID()#',this.value,\'STRING\');" /> ' );
+              }
               if (rgbw_controller != -1) {
 
                 if (iV == "0") {
@@ -529,9 +541,11 @@
               && (oucfm == -1)
               && (oucfmTW == -1)
               && (oucmpcb == -1)
+              && (oump3led == -1)
               && (partyDialogDevice == -1)
               && (statusDisplayDevice == -1)
               && (statusDisplayEPaperDevice == -1)
+              && (statusHbDisplayEPaperDevice == -1)
               && (rgbw_controller == -1)
               && (jalousieActor == -1)
               && (i_vir_lg_rgb_dim == -1)
@@ -546,6 +560,9 @@
               if((oucfm != -1) || (oucfmTW != -1) || (oucmpcb != -1)) {
                 Write( '<span class="CLASS02401" onclick="ChangeOUCFMValue('#oSD.ID()#');"><img src="/ise/img/notepad.png" /></span>' );
               }
+              if(oump3led != -1) {
+                Write( '<span class="CLASS02401" onclick="ChangeHBOUCFMValue('#oSD.ID()#');"><img src="/ise/img/notepad.png" /></span>' );
+              }              
               if (partyDialogDevice != -1) {
                 Write( '<span id="setPartyMode" class="CLASS02401" onclick="SetPartyMode(this);"><img src="/ise/img/notepad.png" /></span>' );
               }
@@ -555,6 +572,9 @@
               if (statusDisplayEPaperDevice != -1) {
                 Write( '<span id="setStatusDisplayEPaper'#oSD.ID()#'" class="CLASS02401" onclick="SetStatusDisplay('#oSD.ID()#',\'DIS-EP\');"><img src="/ise/img/notepad.png" /></span>' );
               }
+              if (statusHbDisplayEPaperDevice != -1) {
+                Write( '<span id="setHbStatusDisplayEPaper'#oSD.ID()#'" class="CLASS02401" onclick="SetStatusDisplay('#oSD.ID()#',\'DIS-EP42BW\');"><img src="/ise/img/notepad.png" /></span>' );
+              }
               if (rgbw_controller != -1) {
                 Write( '<span id="setRGBWController'#oSD.ID()#'" class="CLASS02401" onclick="SetRGBWController('#oSD.ID()#',\''#oDP.HSSID()#'\');"><img src="/ise/img/notepad.png" /></span>' );
               }
